presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraClientModule.java

```diff
@@ -161,6 +161,10 @@ public static CassandraSession createCassandraSession(
             ));
         }
 
-        return new NativeCassandraSession(connectorId.toString(), extraColumnMetadataCodec, clusterBuilder.build(), config.getNoHostAvailableRetryTimeout());
+        return new NativeCassandraSession(
+                connectorId.toString(),
+                extraColumnMetadataCodec,
+                new ReopeningCluster(clusterBuilder::build),
```

Comment:
Commit title ("Reopen cluster after failure") and message are ambiguous when read without PR context (e.g. in `git log`)